samples/test-samples.cmake

```diff
@@ -38,9 +38,18 @@ foreach (SAMPLE data-sealing)
     COMMAND ${CMAKE_COMMAND} --build ${SOURCE_DIR}/${SAMPLE}
     WORKING_DIRECTORY ${SAMPLE_BUILD_DIR})
 
-  execute_process(
-    COMMAND ${CMAKE_COMMAND} --build ${SAMPLE_BUILD_DIR} --target run
-    RESULT_VARIABLE TEST_RESULT)
+  if (NOT DEFINED $ENV{OE_SIMULATION})
+    execute_process(
+      COMMAND ${CMAKE_COMMAND} --build ${SAMPLE_BUILD_DIR} --target run
+      RESULT_VARIABLE TEST_RESULT)
+  endif ()
+
+  if (${SAMPLE} MATCHES "(file-encryptor|helloworld)")
+    # Simulation tests
+    execute_process(
+      COMMAND ${CMAKE_COMMAND} --build ${SAMPLE_BUILD_DIR} --target simulate_run
+      RESULT_VARIABLE TEST_RESULT)
```

Comment:
This will overwrite the result of the variable above, leading to missed failures if the non-simulation run fails, but the simulation runs.

Reply:
Oops, thanks for catching this one, I missed it.